### ISSUE TITLE
CreateOrConnect failing with ambiguous error

### ISSUE THREAD

**Issue by marytal** [prisma/prisma#16183]
### Bug description

I am attempting to use `connectOrCreate` inside of a prisma `update` call. 
Although everything seems to be correct, I'm getting an ambiguous error that I can't quite understand. 

```json
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": "P2025",
          "clientVersion": "3.12.0",
          "meta": {
            "cause": "No 'CohortStudent' record (needed to inline the relation the update for 'Student' record(s)) was found for a nested connect or create on one-to-many relation 'CohortStudentToStudent'."
          },
```

Isn't the whole point of using `connectOrCreate` that no record has to exist? That one will be created if there isn't one already?

### How to reproduce

1. Create a 1 to many relationship between two tables (i.e. tweets have likes)
2. Create a record that does has zero items of the relation (i.e. tweet with no likes)
3. Attempt to update that record, passing in an array of `connectOrCreate` "like" records into the tweet update call
4. You should see an error saying "No 'Like' record (needed to inline the relation the update for 'Tweet' record(s)) was found for a nested connect or create on one-to-many relation 'TweetToLike'."

### Expected behavior

You should create new tweet records if the ones you have passed don't exist, otherwise you should connect those likes to the tweet. 

### Prisma information

```prisma
model Student {
  id             Int      @id @default(autoincrement())
  createdAt      DateTime @default(now())
  updatedAt      DateTime @default(now()) @updatedAt
  externalId     String
  organizationId Int
  email          String?  @db.Citext
  fullName       String?
  notes          String?

  cohortStudents CohortStudent[]

  @@unique([externalId, organizationId])
}

model CohortStudent {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  studentId Int
  cohortId  Int

  //relations
  student Student @relation(fields: [studentId], references: [id])
  cohort  Cohort  @relation(fields: [cohortId], references: [id])

  @@unique([studentId, cohortId])
}
```

```ts
function updateStudentByExternalId(input: UpdateStudentByExternalIdInput) {
  const {
    studentId,
    externalId,
    organizationId,
    cohorts,
    newExternalId,
    email,
    fullName,
  } = input;

  const cohortConnectionMutationData = cohorts.map((cohort) => {
    return {
      create: {
        cohortId: parseInt(cohort.cohortId),
      },
      where: {
        studentId_cohortId: {
          cohortId: parseInt(cohort.cohortId),
          studentId: parseInt(studentId),
        },
      },
    };
  });

  return prisma.student.update({
    where: {
      externalId_organizationId: {
        externalId: externalId,
        organizationId: parseInt(organizationId),
      },
    },
    data: {
      externalId: newExternalId,
      email,
      fullName,
      cohortStudents: {
        connectOrCreate: cohortConnectionMutationData,
      },
    },
  });
}
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
prisma                  : 3.12.0
@prisma/client          : 3.12.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 22b822189f46ef0dc5c5b503368d1bee01213980
Studio                  : 0.459.0
```

**Comment by SevInf** [comment]
Hi @marytal!
Could you try more recent prisma version please? We fixed quite a few issues since `3.12.0`.

**Comment by marytal** [comment]
Updating to the latest version fixed it :) Thanks for the tip!
